### PR TITLE
Add Supply.first

### PR DESCRIPTION
**Pull request by stoned**
Document Supply.first. Ref #3187

### CODE REVIEW

File: doc/Type/Supply.pod6 [hunk 1]
```diff
@@ -370,6 +370,22 @@ Creates a "tail" supply with the same semantics as L<List.tail|/type/List#method
     my $ts = $s.tail(2);
     $ts.tap(&say);           # OUTPUT: «3␤2␤»
 
+=head2 method first
+
+    method first(Supply:D: :$end, |c)
+
+This method is a frontend on the C<grep> method. Its creates a new "head"
```
Comment (JJ) [inline doc/Type/Supply.pod6]: Not clear what you mean here. If it means that it's like getting a supply of the first element that would be returned from grep, say so. Saying it's a frontend for something is like saying you've put a layer over that something, but that it does the same, maybe in a different way.

File: doc/Type/Supply.pod6 [hunk 2]
```diff
@@ -370,6 +370,22 @@ Creates a "tail" supply with the same semantics as L<List.tail|/type/List#method
...
+supply, or a "tail" supply if the optional named parameter C<:end>
+is truthy, from a supply created by calling the C<grep> method on the
```
Comment (JJ) [inline doc/Type/Supply.pod6]: Maybe you mean something like "bookend"? It returns either the first or the last element that meets that expression. Or ot emits the first of the remaining elements?

File: doc/Type/Supply.pod6 [hunk 3]
```diff
@@ -370,6 +370,22 @@ Creates a "tail" supply with the same semantics as L<List.tail|/type/List#method
...
+invocant, with any remaining arguments as parameters.
+If there are no remaining argument, this method is equivalent to calling
+on the invocant, without parameter, the C<head> or the C<tail> method,
+according to named parameter C<:end>.
+
+    my $rand = supply { emit (rand × 100).floor for ^∞ };
+    my $first-prime = $rand.first: &is-prime;
+    $first-prime.tap: &say; # output the first prime from the endless random number supply $rand
+
```
Comment (JJ) [inline doc/Type/Supply.pod6]: maybe add "and then stop... "

**Comment by stoned** [comment]
Thanks JJ for the suggestions! I pushed a new version. Please let me know what you think of it.

**Comment by JJ** [comment]
> Thanks JJ for the suggestions! I pushed a new version. Please let me know what you think of it.

Looks fine! Waiting for the CI result to merge.

**Pull request merged**